File: cs205/DAA_1&2/py_programs/LSW.py
import datetime

def findval(A,val):
    n = len(A)
    for i in range(n):
        if A[i]==val:
            return i
    return -1

print("Worst Case: Element does not exist in array")
for i in range(1,11):
    print("File",i)
    file = open("..\\..\\Files\\File "+str(i)+".txt","r")
    A = list(map(int,file.read().split()))
    file.close()
    Ttime = 0
    for i in range(100):
        start = datetime.datetime.now()
        X = findval(A,-1)
        # if X!=-1:
        #     print("Element found at index",X)
        # else:
        #     print("Element not found in given array")
        end = datetime.datetime.now()
        Ttime += float((end-start).total_seconds()*1000)
    print("Worst case average time:",Ttime/100,"milliseconds")

File: cs205/DAA_1&2/cpp_programs/LSW.cpp
#include<bits/stdc++.h>
using namespace std;

long long int findval(vector<int>& A, int val){
    long long int n = A.size();
    for(int i=0;i<n;i++){
        if(A[i]==val) return i;
    }
    return -1;
}

int main(){
    cout<<"Worst Case: element does not exist in array"<<endl;
    for(int i=0; i<10; i++){
        ifstream file("..\\..\\Files\\File "+to_string(i+1)+".txt");
        vector<int> numbers;
        int num;
        while(file>>num){
            numbers.push_back(num);
        }
        cout<<"File "<<i+1<<endl;
        long long int time = 0;
        for(int i=0;i<100;i++)
        {
            auto start = chrono::high_resolution_clock::now();
            findval(numbers,-1);
            auto end = chrono::high_resolution_clock::now();
            chrono::duration<double,milli>ctime = end-start;
            time += ctime.count();
        }
        cout<<"Average time: "<<time/100.0<<"ms"<<endl;
    }
}